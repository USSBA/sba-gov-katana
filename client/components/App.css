.app {
    background-color: yellow;
    height: 50px;
    border: 1px solid blue;
}
