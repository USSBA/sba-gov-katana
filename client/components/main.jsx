import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux'
import nameReducer from '../reducers/name.js'
import LincMain from './linc-main.jsx';

const store = createStore(nameReducer);


ReactDOM.render(
  ( <Provider store={store}>
      <LincMain />
    </Provider>),
  document.getElementById('root')
);