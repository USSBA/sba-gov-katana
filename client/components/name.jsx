import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as NameActions from '../action-creators/name.js';

class NameInput extends React.Component {
    static propTypes = {
        actions: React.PropTypes.object.isRequired
    }
    doStuff(event){
        let value = event.target.value;
        this.props.actions.updateName(num);
    }
    render(){
        return (
            <div>
            <ul>
                {this.props.names.map((n) => (<li>n</li>))}
            </ul>
            <input 
                type="text"
                value={this.props.name}
                onChange={this.doStuff}
            />   
            </div>
        );
    }
}

function mapStateToProps(state) {
  console.log("The state is " + JSON.stringify(state));
  return {
    name: state.name,
    names: state.names
  };
}

function mapDispatchToProps(dispatch) {
  return{
    actions: bindActionCreators(NameActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NameInput);



