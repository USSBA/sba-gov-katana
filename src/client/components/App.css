.app {
    background-color: yellow;
    height: 50px;
    border: 1px solid blue;
}

.hidden{
    display: none;
}

body{
    background-color: grey;
}



