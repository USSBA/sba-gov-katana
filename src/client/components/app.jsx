import React from 'react';
import { Provider } from 'react-redux';
import { applyMiddleware, createStore } from 'redux';
import thunk from "redux-thunk";
import rootReducer from '../reducers/index.js'
import { logPageView, googleAnalyticsMiddleware } from "../services/analytics.js";
import logging from "../services/logger.js";

import { Route, browserHistory, IndexRoute, IndexRedirect } from 'react-router';
import { syncHistoryWithStore } from 'react-router-redux';
import HaxRouter from './hax-router.jsx';

import Themer from './templates/themer/themer.jsx'


let middlewareList = [];
middlewareList.push(thunk);
if (window.CONFIG.debug) {
  middlewareList.push(logging);
}
if (window.CONFIG.googleAnalytics.enabled) {
  middlewareList.push(googleAnalyticsMiddleware);
}

const middleware = applyMiddleware(...middlewareList);
const store = createStore(rootReducer, middleware);

// Create an enhanced history that syncs navigation events with the store
const history = syncHistoryWithStore(browserHistory, store);

import '../styles/base.scss'
require('font-awesome-webpack2');


class App extends React.Component {
  render() {
    return (
      <Provider store={ store }>
        <Themer>
          <HaxRouter onUpdate={ logPageView() } history={ history }>
            { this.props.routes }
          </HaxRouter>
        </Themer>
      </Provider>
      );
  }
}

export default App;
