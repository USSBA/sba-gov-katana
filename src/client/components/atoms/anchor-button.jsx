import React from 'react';
import { Button } from 'react-bootstrap'

class AnchorButton extends React.Component {

  scroll() {
    let element = document.getElementById(this.props.anchor);
    element.scrollIntoView({
      block: "start",
      behavior: "smooth"
    });
  }

  render() {
    return (
      <Button block className={ this.props.className } onClick={ this.scroll.bind(this) }>
        {this.props.text}
      </Button>
    )
  }
}

export default AnchorButton;
