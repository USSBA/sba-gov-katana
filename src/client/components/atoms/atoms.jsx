import CaptionText from "./caption-text"
