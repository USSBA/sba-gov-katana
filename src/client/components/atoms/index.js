import ArrowButton from "./arrow-button/arrow-button.jsx";
import ButtonBase from "./button-base/button-base.jsx";
import CaptionText from "./caption-text/caption-text.jsx";
import Carousel from "./carousel/carousel.jsx";
import Checkbox from "./checkbox/checkbox.jsx";
import DecorativeDash from "./decorative-dash/decorative-dash.jsx";
import ExtraLargeTitleText from "./extra-large-title-text/extra-large-title-text.jsx";
import FormErrorMessage from "./form-error-message/form-error-message.jsx";
import HamburgerIcon from "./hamburger-icon/hamburger-icon.jsx";
import LargeGreySecondaryButton from "./large-grey-secondary-button/large-grey-secondary-button.jsx";
import LargeInversePrimaryButton from "./large-inverse-primary-button/large-inverse-primary-button.jsx";
import LargeInverseSecondaryButton from "./large-inverse-secondary-button/large-inverse-secondary-button.jsx";
import LargePrimaryButton from "./large-primary-button/large-primary-button.jsx";
import LargeSecondaryButton from "./large-secondary-button/large-secondary-button.jsx";
import MainLogo from "./main-logo/main-logo.jsx";
import Multiselect from "./multiselect/multiselect.jsx";
import NewsletterSmallPrimaryButton from "./newsletter-small-primary-button/newsletter-small-primary-button.jsx";
import PageLink from "./page-link/page-link.jsx";
import PdfIcon from "./icons/pdf.jsx";
import ProgressBar from "./progress-bar/progress-bar.jsx";
import Radio from "./radio/radio.jsx";
import SectionLink from "./section-link/section-link.jsx";
import SimpleSelect from "./simple-select/simple-select.jsx";
import SmallGreySecondaryButton from "./small-grey-secondary-button/small-grey-secondary-button.jsx";
import SmallGreySecondaryFormButton from "./small-grey-secondary-form-button/small-grey-secondary-form-button.jsx";
import SmallIcon from "./small-icon/small-icon.jsx";
import SmallInversePrimaryButton from "./small-inverse-primary-button/small-inverse-primary-button.jsx";
import SmallInverseSecondaryButton from "./small-inverse-secondary-button/small-inverse-secondary-button.jsx";
import SmallPrimaryButton from "./small-primary-button/small-primary-button.jsx";
import SmallPrimaryFormButton from "./small-primary-form-button/small-primary-form-button.jsx";
import SmallSecondaryButton from "./small-secondary-button/small-secondary-button.jsx";
import SocialMediaLink from "./social-media-link/social-media-link.jsx";
import SubtitleText from "./subtitle-text/subtitle-text.jsx";
import TextInput from "./text-input/text-input.jsx";
import TextArea from "./textarea/textarea.jsx";
import UtilityLink from "./utility-link/utility-link.jsx";
import DocumentType from "./document-type/document-type.jsx";
import ValidationIcon from "./validation-icon/validation-icon.jsx";
import SearchIcon from "./icons/search.jsx";
import ApplyButton from "./apply-button/apply-button.jsx";
import CaretIcon from "./icons/caret-down.jsx";
import CloseIcon from "./icons/close.jsx";
import HomeIcon from "./icons/home.jsx";
import FailureIcon from "./icons/failure.jsx";
import SuccessIcon from "./icons/success.jsx";

export { ArrowButton, ButtonBase, CaptionText, Carousel, Checkbox, DecorativeDash, ExtraLargeTitleText, FormErrorMessage, HamburgerIcon, LargeGreySecondaryButton, LargeInversePrimaryButton, LargeInverseSecondaryButton, LargePrimaryButton, LargeSecondaryButton, MainLogo, Multiselect, NewsletterSmallPrimaryButton, PageLink, PdfIcon, ProgressBar, Radio, SectionLink, SimpleSelect, SmallGreySecondaryButton, SmallIcon, SmallInversePrimaryButton, SmallInverseSecondaryButton, SmallPrimaryButton, SmallSecondaryButton, SocialMediaLink, SubtitleText, TextInput, TextArea, UtilityLink, ValidationIcon, SearchIcon, ApplyButton, DocumentType, CaretIcon, CloseIcon, HomeIcon, FailureIcon, SuccessIcon };
