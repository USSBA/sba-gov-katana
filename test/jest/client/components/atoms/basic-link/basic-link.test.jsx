/* eslint-disable no-unused-vars,no-undef */
import React from "react";
import {shallow} from "enzyme";
import renderer from "react-test-renderer";
import _ from "lodash";
/*eslint-enable no-unused-vars*/

// Quiet warnings about OnTouchTap
import injectTapEventPlugin from "react-tap-event-plugin";
injectTapEventPlugin();

import {BasicLink} from "client/components/atoms";

describe("BasicLink", () => {
  describe("normal link", () => {
    const defaultProps = {
      url: "/hello/world",
      text: "Hello, World!"
    };
    test("Renders as expected", () => {
      const component = renderer.create(<BasicLink {...defaultProps} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    test("Renders a button element", () => {
      const component = shallow(<BasicLink {...defaultProps}/>);
      expect(component.type()).toEqual("a");
    });
    test("Sets the element class", () => {
      const className = "cheeseburger"
      const componentProps = _.merge(_.clone(defaultProps), {myClassName: className});
      const component = shallow(<BasicLink {...componentProps}/>);
      expect(component.props().className).toEqual(className);
    });
    test("Sets arbitrary html properties on the element", () => {
      const customStyle = "color:green";
      const componentProps = _.merge(_.clone(defaultProps), {style: customStyle});
      const component = shallow(<BasicLink {...componentProps}/>);
      expect(component.props().style).toEqual(customStyle);
    });
    test("Assigns a provided onClick to onTouchTap", () => {
      const customOnClick = () => {return "myOnClick";};
      const componentProps = _.merge(_.clone(defaultProps), {onClick: customOnClick});
      const component = shallow(<BasicLink {...componentProps}/>);
      expect(component.props().onTouchTap()).toEqual("myOnClick");
    });

  });
  describe("button link", () => {
    const defaultProps = {
      url: "/hello/world",
      text: "Hello, World!",
      htmlElement: "button"
    };
    test("Renders as expected", () => {
      const component = renderer.create(<BasicLink {...defaultProps}/>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    test("Renders a button element", () => {
      const component = shallow(<BasicLink {...defaultProps}/>);
      expect(component.type()).toEqual("button");
    });
  });
});