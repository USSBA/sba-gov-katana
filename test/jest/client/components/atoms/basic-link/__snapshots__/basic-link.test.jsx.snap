// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLink button link Renders as expected 1`] = `
<button
  autoFocus={false}
  className=""
  onKeyPress={[Function]}
  onTouchTap={[Function]}
  tabIndex="0"
>
  Hello, World!
</button>
`;

exports[`BasicLink normal link Renders as expected 1`] = `
<a
  autoFocus={false}
  className=""
  onKeyPress={[Function]}
  onTouchTap={[Function]}
  tabIndex="0"
>
  Hello, World!
</a>
`;
